mpt/crawler.py

```diff
@@ -0,0 +1,111 @@
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.common.exceptions import NoSuchElementException
+from selenium.common.exceptions import StaleElementReferenceException
+
+# Constants
+BASE_URL = "https://mpt.mp.br/MPTransparencia/pages/portal/"
+TYPES = {
+        0 : "remuneracaoMembrosAtivos",
+        1 : "proventosMembrosInativos",
+        2 : "remuneracaoServidoresAtivos",
+        3 : "proventosServidoresInativos",
+        4 : "proventosPensionistas",
+        5 : "proventosColaboradores"
+    }
+MONTHS = {
+        '1' : "Janeiro",
+        '2' : "Fevereiro",
+        '3' : "Março",
+        '4' : "Abril",
+        '5' : "Maio",
+        '6' : "Junho",
+        '7' : "Julho",
+        '8' : "Agosto",
+        '9' : "Setembro",
+        '10' : "Outubro",
+        '11' : "Novembro",
+        '12' : "Dezembro"
+    }
+
+# Retrieves payment files from MPT
+def crawl(output_path, month, year):
+    files = []
+    urls = links()
+
+    # Creates the download directory (if it doesn't exists)
+    pathlib.Path('./' + output_path).mkdir(exist_ok = True)
+    
+    for i, url in zip(TYPES.keys(), urls):
+        file_name = TYPES[i] + '-' + month + "-" + year + '.ods'
+        
+        file_path = (output_path + "/" + file_name)
+        download(url, output_path, month, year)
+        files.append(file_path)
+
+    return files
+
+def links():
+    links = []
+    for i in range(len(TYPES)):
+        links.append(BASE_URL + TYPES[i] + ".xhtml")
+    return links
+
+def download(url, file_path, month, year):
+    driver = setup_driver()
+    driver.get(url)
+    years = driver.find_element_by_id("j_idt140")
+    now = datetime.datetime.now()
+    current_year = now.year
+    
+    for _ in range(current_year - int(year)):
+        years.send_keys(Keys.ARROW_DOWN)
+    try:
+        WebDriverWait(driver, 10)
+        search = driver.find_element_by_link_text("Pesquisar")
+        search.click()
+    except NoSuchElementException:
+        try:
+            search = driver.find_element_by_name("j_idt143")
+            search.click()
+        finally:
+            print("Ano identificado")
+
+    id_month = "tabelaRemuneracao:" + str(int(month) - 1) + ":j_idt157"
+    try:
+        archive = WebDriverWait(driver, 10).until(
+            EC.presence_of_element_located((By.ID, id_month))
+        )
+        try:
+            archive.send_keys("\n")
+            driver.execute_script("arguments[0].click();", archive)
+            WebDriverWait(driver, 10)
+        finally:
+            print("Mês identificado")
+    finally:
+        print("Download efetuado")
+    driver.quit()
+
+def setup_driver():
+    env_path = '.env'
+    load_dotenv(dotenv_path = env_path)
+    current_directory = os.getcwd()
+
+    RELATIVE_PATH_CHROME = os.getenv("PATH_DRIVER")
```

Comment:
Você também deve validar esse parâmetro. Ou seja, tudo que o usuário `deve` passar para usar seu coletor tem que ser checado e essa checagem tem que ser feita no main

Reply:
Nesse caso o path_driver deve ser o caminho até o chromedriver. Ainda não sei bem como isso vai ser definido, mas acredito que não vai ser necessário o usuário informar sempre essa direção, uma vez que no docker eu só vou instalar uma vez, e o caminho não vai mudar. Posso só fazer algo do tipo "path_driver = "/chromedriver" ", no próprio método setup_driver, ou defino isto no main e envio como parametro, ou nenhum destes?

Reply:
Para o seu coletor, isso vai ser um caminho passado via variável de ambiente. O main vai checar essa variável e dar erro se ela não for passada. A partir daí, dentro do programa ela é passada via parâmetros das funções. Isso torna a leitura do fluxo de execução muito mais fácil. Variáveis de ambiente são parâmetro, porém não sãop declarados no cabeçalho (o que é ruim nesse caso).

Quando o coletor estiver num container, o próprio Dockerfile vai setar essa variável (não vai ser o dev)